### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by pindamonhangaba**
More descriptive section on React integration. Separated code samples and highlighted Cursor registration with functions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -306,8 +306,10 @@ This basically makes the `shouldComponentUpdate` method useless in most of cases
 
 ##### Tree level
 
-You can bind a React component to the tree itself and register some handy cursors:
+You can bind a React component to a Baobab tree and register one or more cursors. Cursors registered to a component will have their contents bound to the `this.state.cursor` (or `this.state.cursors` if registering multiple cursors).
 
+###### Single cursor binding:
+Registering a cursor is as simple as returning a path.
```
Comment (Yomguithereal) [inline README.md]: I am not sure `returning` is the correct word here.

File: README.md [hunk 2]
```diff
@@ -316,107 +318,134 @@ var tree = new Baobab({
...
   render: function() {
-    var renderItem = function(name) {
-      return <li>{name}</li>;
-    };
-
     // Cursor data is then available either through:
     var data = this.cursor.get();
     // Or
-    var data = this.state.cursor;
+    data = this.state.cursor;
```
Comment (Yomguithereal) [inline README.md]: You forgot `var` here to keep consistent with other examples.

File: README.md [hunk 3]
```diff
@@ -316,107 +318,134 @@ var tree = new Baobab({
...
 
-// Multiple cursors
-var UserList = React.createClass({
-  mixins: [tree.mixin],
-  cursors: [['users'], ['information', 'title']],
-  render: function() {
-    var renderItem = function(name) {
-      return <li>{name}</li>;
-    };
+You can also use a function returning a Cursor. This allows the component to be bound to a cursor passed trought `this.props`.
```
Comment (Yomguithereal) [inline README.md]: `through` rather.

**Comment by Yomguithereal** [comment]
Thanks for the input @pindamonhangaba! I added some comments on your changes. Else that this, your examples are clearer than mine and shall be merged asap.

**Comment by pindamonhangaba** [comment]
Ok, I'm not sure how to proceed on fixing what you pointed out, do I fix it here and create another pull request?

**Comment by Yomguithereal** [comment]
You can just push a new commit on your fork and the PR shall adapt automatically. What would be top notch would be to rebase your PR so that it is composed of only one atomic commit.

**Comment by pindamonhangaba** [comment]
Welp, tried to rebase, not sure if worked.

File: README.md [hunk 4]
```diff
@@ -316,107 +318,136 @@ var tree = new Baobab({
...
     // Or
     var data = this.state.cursor;
 
-    return <ul>{this.cursor.get().map(renderItem)}</ul>;
+    return <ul>{data.map(function(username) {
+        return <li>{username}</li>;
+    })}</ul>;
   }
 });
+cursor: ['usergroups', 'group1'],
```
Comment (Yomguithereal) [inline README.md]: What are those lines?

**Comment by Yomguithereal** [comment]
Hello @pindamonhangaba, try to squash the last commits into the first one. [This](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#squash-your-changes-optional) page describes how to do it if I recall correctly :).

**Comment by pindamonhangaba** [comment]
I had tried that, but I got carried away and didn't realize I should've just rebased master.

**Comment by Yomguithereal** [comment]
Thanks @pindamonhangaba. Merging right now.

**Pull request merged**